addons/udes_stock/models/stock_picking.py

```diff
@@ -2784,130 +2784,137 @@ def raise_insufficiency_error(moves):
             to_reserve = picking_type.u_num_reservable_pickings
             reserve_all = to_reserve == -1
             base_domain = [("picking_type_id", "=", picking_type.id), ("state", "=", "confirmed")]
-            limit = 1
+            limit = 100 if reserve_all else picking_type.u_num_reservable_pickings
             processed = Picking.browse()
             by_type = lambda x: x.picking_type_id == picking_type
 
             while reserve_all or to_reserve > 0:
 
                 if self:
                     # Removed processed pickings from self
-                    pickings = self.filtered(by_type) - processed
+                    pickings_to_process = self.filtered(by_type) - processed
                 else:
                     domain = base_domain[:]
                     if processed:
                         domain.append(("id", "not in", processed.ids))
-                    pickings = Picking.search(domain, limit=limit)
+                    pickings_to_process = Picking.search(domain, limit=limit)
 
-                if not pickings:
+                if not pickings_to_process:
                     # No pickings left to process.
                     # If u_num_reservable_pickings is -1, or there are
                     # fewer available pickings that the limit, the loop must
                     # terminate here.
                     break
 
-                batch = pickings.mapped("batch_id")
-                if batch and batch.state == "draft":
-                    # Add to seen pickings so that we don't try to process
-                    # this batch again.
-                    processed |= batch.picking_ids
-                    continue
-
-                if batch and picking_type.u_reserve_batches:
-                    pickings = batch.picking_ids
-
-                # Check for available stock and add pickings which we cannot
-                # fulfill to the processed recordset. The logic here
-                # should mirror that within the savepoint below, expect here we
-                # do not have the move lines.
-                unreservable_moves = pickings._find_unreservable_moves(return_fast=bool(self))
-                if unreservable_moves:
-                    moves = pickings.mapped("move_lines")
-                    moves_todo = (
-                        moves.filtered(lambda m: m.state in ["confirmed", "draft"])
-                        - unreservable_moves
-                    )
-                    if not picking_type.u_handle_partials:
-                        # Add to processed to skip reservation.
-                        processed |= pickings
-
-                        if self:
-                            raise_insufficiency_error(unreservable_moves)
+                # "pickings" is plural as the individual picking may be
+                # extended if it's in a batch.
+                for pickings in pickings_to_process:
```

Comment:
Read the comment but still think is confusing.
Maybe better for picking in pickings_to_process: and after if-s if pickings is not assigned 
pickings = picking somewhere before line 2829 i think

Reply:
I'm not very fond of it either, so I'm willing to go with whatever the team decides.  But I'm not sure I like the idea of changing the name part way through the loop :man_shrugging:

Reply:
I believe that is very obvious that pickings iterator is not the right name

Reply:
Probably we can use `picking` here and then use `to_assign` later on instead of pickings.

Reply:
Applied @ws-mps's suggestion in 8da3aba7